Lib/fontmake/__main__.py

```diff
@@ -161,12 +161,23 @@ def main(args=None):
              '%(choices)s. Default: INFO')
     args = vars(parser.parse_args(args))
 
-    project = FontProject(timing=args.pop('timing'),
-                          verbose=args.pop('verbose'))
-
     glyphs_path = args.pop('glyphs_path')
     ufo_paths = args.pop('ufo_paths')
     designspace_path = args.pop('mm_designspace')
+
+    if 'variable' in args['output']:
+        if not (glyphs_path or designspace_path):
+            parser.error(
+                'Glyphs or designspace source required for variable font')
```

Comment:
Yeah, the info in the UFOs is not sufficient. Even if they have openTypeOS2WeightClass and openTypeOS2WidthClass, that’s not enough to make a VF.

The designspace is a file (see https://github.com/LettError/designSpaceDocument or https://github.com/LettError/MutatorMath or https://github.com/fonttools/fonttools/blob/master/Lib/fontTools/varLib/designspace.py) which fontTools.varLib uses to write the OpenType design-variation space information and which MutatorMath uses to interpolate instance UFOs.